### PR TITLE
feat(LinkedIn Insights Tag): add support for event-specific conversion

### PR DESCRIPTION
**Pull request by sauravlal-rudderstack**
## Description of the change

> support for event-specific conversion

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/648)

### CODE REVIEW

File: integrations/LinkedInInsightTag/browser.js [hunk 1]
```diff
@@ -30,5 +35,32 @@ class LinkedInInsightTag {
...
+    const { message } = rudderElement;
+    const { event } = message;
+    if (!event) {
+      logger.error(
+        "[LinkedIn Insight Tag]: Event name is missing for track call!!==="
+      );
+      return;
+    }
+    const trimmedEvent = event.trim();
+    const eventMapping = getHashFromArray(
```
Comment (utsabc) [inline integrations/LinkedInInsightTag/browser.js:50]: 1-1 mapping is supported or 1 - many is supported. The code is written for 1-1

Comment (sauravlal-rudderstack) [reply integrations/LinkedInInsightTag/browser.js:50]: Yes, 1-1 mapping is supported.

Comment (sauravlal-rudderstack) [reply integrations/LinkedInInsightTag/browser.js:50]: I'll update it with 1 to many.

Comment (sauravlal-rudderstack) [reply integrations/LinkedInInsightTag/browser.js:50]: I'll update it with 1 to many.

File: integrations/LinkedInInsightTag/browser.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 import logger from "../../utils/logUtil";
 import ScriptLoader from "../ScriptLoader";
+import { getHashFromArrayWithDuplicate } from "../utils/commonUtils";
 import { NAME } from "./constants";
 
 class LinkedInInsightTag {
   constructor(config) {
     this.name = NAME;
     this.partnerId = config.partnerId;
+    this.eventConversionId = config.eventConversionId;
```
Comment (ItsSudip) [inline integrations/LinkedInInsightTag/browser.js:11]: Update this variable name to `eventToConversionIdMap`

File: integrations/LinkedInInsightTag/browser.js [hunk 3]
```diff
@@ -18,6 +20,9 @@ class LinkedInInsightTag {
     if (!this.partnerId) {
       return;
     }
+    if (this.eventConversionId === undefined) {
+      this.eventConversionId = [];
+    }
```
Comment (ItsSudip) [inline integrations/LinkedInInsightTag/browser.js:25]: No need to check it here. We are already checking this inside `getHashFromArrayWithDuplicate` function.

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: integrations/LinkedInInsightTag/browser.js (hunk 2 above)
Comment (sauravlal-rudderstack) [reply integrations/LinkedInInsightTag/browser.js]: updated

File: integrations/LinkedInInsightTag/browser.js (hunk 3 above)
Comment (sauravlal-rudderstack) [reply integrations/LinkedInInsightTag/browser.js]: updated

File: integrations/LinkedInInsightTag/browser.js [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 import logger from "../../utils/logUtil";
 import ScriptLoader from "../ScriptLoader";
+import { getHashFromArrayWithDuplicate } from "../utils/commonUtils";
 import { NAME } from "./constants";
 
 class LinkedInInsightTag {
   constructor(config) {
     this.name = NAME;
     this.partnerId = config.partnerId;
+    this.eventToConversionIdMap = config.eventConversionId;
```
Comment (ItsSudip) [inline integrations/LinkedInInsightTag/browser.js:11]:
```suggestion
    this.eventToConversionIdMap = config.eventToConversionIdMap;
```

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: integrations/LinkedInInsightTag/browser.js (hunk 4 above)
Comment (sauravlal-rudderstack) [reply integrations/LinkedInInsightTag/browser.js]: Done

**Review by ItsSudip — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**